index.jsx

```diff
@@ -27,18 +27,42 @@ class Variable extends React.Component {
     return this.props.variable.toUpperCase();
   }
 
+  getSelectedValue(selected) {
+    const { user } = this.props;
+    let selectedValue = {};
+    if (Array.isArray(user.keys)) {
+      selectedValue = selected ? user.keys.find(key => key.name === selected) : user.keys[0];
+    }
+    return selectedValue;
+  }
+
+  // Determining whether to show a dropdown or not onClick
+  // based upon whether the currently displayed value has come
+  // from one of the nested user.keys[] or not.
+  // Additionally do not show the dropdown if there is only a
+  // single key, since there's nothing to change it to.
+  shouldShowVarDropdown(selected) {
+    const { user, variable } = this.props;
+
+    return !!this.getSelectedValue(selected)[variable] && Array.isArray(user.keys) && user.keys.length > 1;
```

Comment:
Should it only count keys with values? Is that possible?
```suggestion
    return !!this.getSelectedValue(selected)[variable] && Array.isArray(user.keys) && user.keys.filter(key => key.name === variable).length > 1;
```

Reply:
We're already checking if this particular variable has a value assigned in the first part of this statement:

```
!!this.getSelectedValue(selected)[variable]
```

Is that sufficient for the use case you're referring to? If there isn't a `user.keys[variable]` then it'll already have exited early from that statement because we already know not to show the var dropdown.

Reply:
Gotcha gotcha, misread that.